modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/deployer/ObjectDefinitionDeployerImpl.java

```diff
@@ -727,6 +636,151 @@ private void _initSystemObjectDefinition(
 			});
 	}
 
+	private boolean _shouldUnregisterApplication(String restContextPath) {
+		List<String> companyIds = _restContextPathCompanyIds.get(
+			restContextPath);
+
+		if ((companyIds != null) && ListUtil.isNotEmpty(companyIds)) {
+			return false;
+		}
+
+		Map<Long, List<ServiceRegistration<?>>> serviceRegistrationsMap =
+			_scopedServiceRegistrationsMap.get(restContextPath);
+
+		if ((serviceRegistrationsMap != null) &&
+			MapUtil.isNotEmpty(serviceRegistrationsMap)) {
+
+			return false;
+		}
+
+		return true;
+	}
+
+	private void _undeployCustomObjectDefinition(
+		ObjectDefinition objectDefinition) {
+
+		long companyId = objectDefinition.getCompanyId();
+		String restContextPath = objectDefinition.getRESTContextPath();
+
+		_undeployObjectDefinitionsMap(companyId, restContextPath);
+		_undeployRestContextPathCompanyIds(companyId, restContextPath);
+		_undeployScopedServiceRegistrationsMap(restContextPath, companyId);
+
+		boolean unregisterApplication = _shouldUnregisterApplication(
+			restContextPath);
+
+		if (unregisterApplication) {
+			_unregisterApplication(unregisterApplication, restContextPath);
```

Comment:
If we want to check to unregister the application before calling the method (I am ok with that, maybe is clearer indeed) I think it is not necessary to pass the boolean `unregisterApplication`. Why? Because here the unregisterApplication will be always true.

Inside the `_unregisterApplication` method, if you check the use of unregisterApplication you will see that it only use here
```
if (!shouldUnregisterApplication) {
	return;
}
```
If unregisterApplication is always true, this `return` statement will never be reached, so, the if is not necessary, which makes unuseful the unregisterApplication argument.

So, to sum up, we can check if it is necessary to unregister the application before calling the method, but in that case, the variable to check if should unregister the application is not necessary inside the `_unregisterApplication` method 🤷 

NOTE: In fact, in Carlos's suggestion about checking this before the unregister method, he proposed to delete the argument in the method
![image](https://user-images.githubusercontent.com/17163902/228634817-04105306-319b-49cb-8278-2d8e5fa148e1.png)